app/auth/middleware.js

```diff
@@ -5,26 +5,33 @@ function _decodeAccessToken (token) {
   return JSON.parse(Buffer.from(payload, 'base64').toString('utf8'))
 }
 
-function processAuthResponse (req, res, next) {
+async function processAuthResponse (req, res, next) {
   const { grant, postAuthRedirect } = req.session
 
   if (!grant) {
     return next()
   }
 
-  req.session.regenerate(error => {
-    if (error) {
-      return next(error)
-    }
-
+  try {
     const accessToken = _decodeAccessToken(grant.response.access_token)
+    const user = new User()
+
+    await user.init(accessToken)
 
-    req.session.authExpiry = accessToken.exp
-    req.session.postAuthRedirect = postAuthRedirect
-    req.session.user = new User(accessToken)
+    req.session.regenerate(async error => {
```

Comment:
Does this require `async`?

Reply:
Good spot. That was from a previous location of this change.